src/instructlab/model/train.py

```diff
@@ -521,20 +615,333 @@ def get_files(directory: str, pattern: str) -> list[str]:
-            )
-            raise click.exceptions.Exit(1)
         train_args, torch_args = map_train_to_library(params)
-        try:
+
+        if strategy == "lab-multiphase":
+            if phased_base_dir and phased_phase1_data and phased_phase2_data:
+                if not skip_user_confirm:
+                    click.confirm(
+                        f"""\n\nSTARTING END-TO-END TRAINING. \nProceeding will OVERWRITE any metadata (checkpoints, intermediate results) that are stored at {phased_base_dir}.\n\nWould you like to proceed?""",
+                        abort=True,
+                    )
+
+                _prepare_phased_base_dir(phased_base_dir)
+
+                _run_phased_training(
+                    ctx=ctx,
+                    train_args=train_args,
+                    torch_args=torch_args,
+                    base_dir=phased_base_dir,
+                    phase1_data=phased_phase1_data,
+                    phase1_num_epochs=phased_phase1_num_epochs,
+                    phase1_samples_per_save=phased_phase1_samples_per_save,
+                    phase2_data=phased_phase1_data,
+                    phase2_num_epochs=phased_phase2_num_epochs,
+                    phase2_samples_per_save=phased_phase2_samples_per_save,
+                    mtbench_judge=phased_mt_bench_judge,
+                )
+
+            else:
+                raise ctx.fail(
+                    "End-to-end training minimally requires: `--phased-base-dir`, `--phased-phase1-data`, and `--phased-phase2-data`. One or more wasn't correctly specified."
+                )
+
+        else:
+            if not os.path.isfile(data_path):
+                ctx.fail(
+                    f"Data path must be to a valid .jsonl file. Value given: {data_path}"
+                )
+
             run_training(train_args=train_args, torch_args=torch_args)
-        # unsure what types of exceptions training library returns, this will catch all for now
-        except Exception as exc:
+
+
+def _prepare_phased_base_dir(phased_base_dir: pathlib.Path) -> None:
+    """Adds phase1 and phase2 directories and subdirectories in phased_base_dir.
+
+    Args:
+        phased_base_dir: directory wrapping phase1 and phase2 cached data.
+    """
+
+    logger.debug(f"Phased Training -- Preparing phased base dir: {phased_base_dir}")
+
+    phase1_dir_path = phased_base_dir / "phase1"
+    phase2_dir_path = phased_base_dir / "phase2"
+
+    for p in [phase1_dir_path, phase2_dir_path]:
+        utils.clear_directory(p)
+        _setup_phase_dirs(p)
+
+
+def _setup_phase_dirs(path: pathlib.Path) -> None:
+    """Creates {path}/checkpoints and {path}/eval_cache directories."""
+    # TODO: these sub-directory names are hard-coded here but they
+    # could be parameterized in config.
+    logger.debug(f"Phased Training -- Created phase directories for {path}")
+    ckpt_path = path / "checkpoints"
+    eval_cache_path = path / "eval_cache"
+
+    os.makedirs(ckpt_path)
```

Comment:
for all of these we should check if they exist first, dont want to unnecessarily re-create

Reply:
it's slightly easier to just scrub the tree below the phase cache and recreate the dir. Overhead for UI program is tiny

Reply:
can improve in the future though, certainly.